### PR TITLE
upgrade shap to 0.39.0

### PR DESCRIPTION
**Pull request by imatiach-msft**
upgrade shap to 0.39.0
- officially add python 3.8 support and remove python 3.5 support

### CODE REVIEW

File: python/interpret_community/common/explanation_utils.py [hunk 1]
```diff
@@ -17,8 +17,10 @@
 with warnings.catch_warnings():
     warnings.filterwarnings('ignore', 'Starting from version 2.2.1', UserWarning)
     import shap
-    from shap.common import DenseData
-
+    try:
+        from shap.common import DenseData
+    except ImportError:
+        from shap.utils._legacy import DenseData
```
Comment (gregorybchris) [inline python/interpret_community/common/explanation_utils.py:23]: As discussed in other PR.. We probably shouldn't take a dependency on deprecated interfaces on purpose..

Comment (gregorybchris) [reply python/interpret_community/common/explanation_utils.py:23]: Other PR here: https://github.com/interpretml/interpret-community/pull/370

Comment (imatiach-msft) [reply python/interpret_community/common/explanation_utils.py:23]: we already support other types of data, numpy, pandas, etc.  We also support shap's DenseData type, which is output in some cases (like when using shap.kmeans)

Comment (gregorybchris) [reply python/interpret_community/common/explanation_utils.py:23]: Talked offline and @imatiach-msft agreed to remove Dense data support from interpret-community in a future PR

**Review by gregorybchris — APPROVED** [review]

**Pull request merged**